### PR TITLE
Fix random failure

### PR DESCRIPTION
**Pull request by francisco-dlp**
Fixes randomly failing test Introduced in #1471. @to266, maybe this be a workaround a dask bug?

### CODE REVIEW

File: hyperspy/tests/io/test_blockfile.py [hunk 1]
```diff
@@ -208,7 +208,8 @@ def test_load_to_memory():
 def test_load_readonly():
     s = hs.load(FILE2, lazy=True)
     k = next(filter(lambda x: x.startswith("array-original"),
-                    s.data.dask.keys()))
+                    (key for key in s.data.dask.keys()
```
Comment (to266) [inline hyperspy/tests/io/test_blockfile.py]: Simpler to just incorporate the `isinstance` into the lambda function/filter:
```python
    k = next(filter(lambda x: isinstance(x, str) and x.startswith("array-original"),
                    s.data.dask.keys()))
```

**Review by to266 — COMMENTED** [review]
Sorry did not notice the random failing / not looked at github after submitting. It's not a dask bug by any means, I just forgot that the key could be either a tuple or just a string.

**Comment by francisco-dlp** [comment]
Done, thanks!

**Comment by to266** [comment]
waiting for CI...

**Comment by ericpre** [comment]
Talking about random/intermittent failures, have you noticed [this one](https://ci.appveyor.com/project/hyperspy/hyperspy/build/1.0.1992/job/btco0mvtljfbna4c) happening once in a while?!
The error occurs when running `test_hologram_image.py` on appveyor 32 bits and the error says "Command exited with code -1073741819".

Is it different from the error you are talking about?

**Comment by francisco-dlp** [comment]
Yes, that's a different one.

**Comment by to266** [comment]
To me it seems that the appveyor simply times out or something along the lines (isn't it always 14'59"?).

Anyway, merging this one

**Pull request merged**